### PR TITLE
Ensure relationship keys are de-camelized in write payload

### PR DESCRIPTION
**Pull request by jplhomer**
We're seeing an issue where a multi-word relationship key (in Rails) is receiving a write payload with a camelized key. The write fails in this case:

```js
{
  data: {
    id: "1",
    relationships: {
      specialBooks: {
        data: {} // etc
      }
    }
}
```

This PR adds a test for this behavior, and introduces the `decamelize` string helper.

Definitely open to more elegant ways to do this without decamelizing a string!

### CODE REVIEW

File: test/integration/nested-persistence-test.ts [hunk 1]
```diff
@@ -199,14 +224,16 @@ describe('nested persistence', function() {
...
       instance.books = [book];
+      instance.specialBooks = [specialBook];
     });
 
     // todo test on the way back - id set, attrs updated, isPersisted
     // todo remove #destroy? and just save when markwithpersisted? combo? for ombined payload
     // todo test unique includes/circular relationshio
     it('sends the correct payload', function(done) {
-      instance.save({ with: { books: 'genre' } }).then((response) => {
+      instance.save({ with: { books: 'genre', specialBooks: true } }).then((response) => {
```
Comment (richmolj) [inline test/integration/nested-persistence-test.ts]: Super minor, but could we make this `specialBooks: {}`? It better models what we actually expect (and will enforce in TypeScript soon), the `true` just *happens* to work here and these tests can be useful as documentation.

Comment (jplhomer) [reply test/integration/nested-persistence-test.ts]: Ahh yes - was honestly just a guess on my part 😄  updated!

**Comment by richmolj** [comment]
@jplhomer one minor comment but this is fantastic, thank you 👏 ! I find it unbelievable I have not come across this before, perhaps one of the minor bumps along the way introduced this. At least now we have a spec in the right place. Very much appreciated! ❤️ 

(I know @wadetandy has had better luck importing lodash functions in TypeScript for things like `decamelize`, but prior to merging the 1.x branch they are a bit of a pain in the ass, so I think the explicit function is fine for now)

**Comment by richmolj** [comment]
Released in `0.5.5`

**Pull request merged**